### PR TITLE
FIX-#6114: Fix series groupby list of numpy methods

### LINKED ISSUE #6114: BUG: series groupby list of numpy methods gives wrong column names
This is upstreaming https://github.com/ponder-org/modin-public/commit/c3013cdc8fa97d7fb5dcc7c99cb5a9d8e67d55a7

I don't have the error on hand right now.

### PR DESCRIPTION
**Pull request by mvashishtha**
## What do these changes do?

upstreaming https://github.com/ponder-org/modin-public/commit/c3013cdc8fa97d7fb5dcc7c99cb5a9d8e67d55a7

Signed-off-by: mvashishtha <mahesh@ponder.io>

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6114
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/groupby.py [hunk 1]
```diff
@@ -711,7 +705,6 @@ def do_relabel(obj_to_relabel):
                     result.columns = new_columns_idx
                     return result
 
-            func_dict = {col: try_get_str_func(fn) for col, fn in func_dict.items()}
```
Comment (RehanSD) [inline modin/pandas/groupby.py:714]: I think we still need this line right - we just have to add the underscore.

**Review by RehanSD — CHANGES_REQUESTED** [review]
Left a quick comment!

File: modin/pandas/groupby.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/groupby.py:714]: Nope, it turns out we don't need it

**Review by RehanSD — DISMISSED** [review]
LGTM!

**Review by RehanSD — APPROVED** [review]
LGTM!

**Pull request merged**